Objects/typeobject.c

```diff
@@ -8493,6 +8542,135 @@ slot_tp_finalize(PyObject *self)
     PyErr_SetRaisedException(exc);
 }
 
+typedef struct _PyBufferWrapper {
+    PyObject_HEAD
+    PyObject *mv;
+    PyObject *obj;
+} PyBufferWrapper;
+
+static int
+bufferwrapper_traverse(PyObject *self, visitproc visit, void *arg)
```

Comment:
Why not just accept self as `PyBufferWrapper ` in the first place? It will look cleaner and avoid another cast.

Reply:
That creates a warning, the param type must be `PyObject *`.

```
Objects/typeobject.c:8591:20: warning: incompatible function pointer types initializing 'traverseproc' (aka 'int (*)(struct _object *, int (*)(struct _object *, void *), void *)') with an expression of type 'int (PyBufferWrapper *, visitproc, void *)' (aka 'int (struct _PyBufferWrapper *, int (*)(struct _object *, void *), void *)') [-Wincompatible-function-pointer-types]
    .tp_traverse = bufferwrapper_traverse,
                   ^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
You can cast the fp with `traverseproc`